securesystemslib/metadata.py

```diff
@@ -52,6 +52,16 @@ def default_deserializer() -> BaseDeserializer:
     def default_serializer() -> BaseSerializer:
         return JSONSerializer()
 
+    @classmethod
+    def from_bytes(
+        cls,
+        data: bytes,
+        deserializer: Optional[BaseDeserializer] = None
+    ) -> "Envelope":
+
+        json = super().from_bytes(data, deserializer)
+        return cls.from_dict(json)
```

Comment:
I think you are mixing concerns here. Shouldn't we expect the deserializer passed to `from_bytes` to return an `Envelope`? IMO it's better and more flexible to move the logic here to a custom:

```python
class EnvelopeJSONDeserializer: # could also inherit from `JsonDeserializer` and call `super()` below
    def deserialize(data):
        return Envelope.from_dict(
            JsonDeserializer().deserialize(data)
        )
```

Then you can return that from `default_deserializer()` and it will provide you with the `from_bytes()` you want without the need to override the one inherited from `SerializationMixin`

Note that in-toto won't use this `EnvelopeJSONDeserializer` because there we want something like an `AnyMetadataJSONDeserializer`, which can return either an `Envelope` or `Metablock` as described in ITE-5. 

So for the purpose of ITE-5, you don't even have to create a `EnvelopeJSONDeserializer`. But I think it's okay to provide one in `securesystemslib` as blueprint for custom deserializers.

What do you think?